support-workers/src/typescript/model/stateSchemas.ts

```diff
@@ -0,0 +1,159 @@
+import { z } from 'zod';
+import { addressSchema } from './address';
+import { paymentFieldsSchema, paymentProviderSchema } from './paymentFields';
+import { paymentMethodSchema } from './paymentMethod';
+import { productTypeSchema } from './productType';
+
+export const titleSchema = z.union([
+	z.literal('Mr'),
+	z.literal('Mrs'),
+	z.literal('Ms'),
+	z.literal('Miss'),
+	z.literal('Mx'),
+	z.literal('Dr'),
+	z.literal('Prof'),
+	z.literal('Rev'),
+]);
+
+export const userSchema = z.object({
+	id: z.string(),
+	primaryEmailAddress: z.string(),
+	title: titleSchema.nullable(),
+	firstName: z.string(),
+	lastName: z.string(),
+	billingAddress: addressSchema,
+	deliveryAddress: addressSchema.nullable(),
+	telephoneNumber: z.string().nullable(),
+	isTestUser: z.boolean().default(false),
+	deliveryInstructions: z.string().nullable(),
+});
+export type User = z.infer<typeof userSchema>;
+
+export const analyticsInfoSchema = z.object({
+	isGiftPurchase: z.boolean(),
+	paymentProvider: paymentProviderSchema,
+});
+
+const ophanIdsSchema = z.object({
+	pageviewId: z.string().nullable(),
+	browserId: z.string().nullable(),
+});
+
+const abTestSchema = z.object({
+	name: z.string(),
+	variant: z.string(),
+});
+
+const queryParamSchema = z.object({
+	name: z.string(),
+	value: z.string(),
+});
+
+const referrerAcquisitionDataSchema = z.object({
+	campaignCode: z.string().nullable(),
+	referrerPageviewId: z.string().nullable(),
+	referrerUrl: z.string().nullable(),
+	componentId: z.string().nullable(),
+	componentType: z.string().nullable(),
+	source: z.string().nullable(),
+	abTests: z.array(abTestSchema).nullable(),
+	queryParameters: z.array(queryParamSchema).nullable(),
+	hostname: z.string().nullable(),
+	gaClientId: z.string().nullable(),
+	userAgent: z.string().nullable(),
+	ipAddress: z.string().nullable(),
+	labels: z.array(z.string()).nullable(),
+});
+
+const acquisitionDataSchema = z.object({
+	ophanIds: ophanIdsSchema,
+	referrerAcquisitionData: referrerAcquisitionDataSchema,
+	supportAbTests: z.array(abTestSchema),
+});
+
+export const giftRecipientSchema = z.object({
+	title: titleSchema.nullable(),
+	firstName: z.string(),
+	lastName: z.string(),
+	email: z.string().nullable(),
+});
+
+const baseStateSchema = z.object({
+	requestId: z.string(),
+	user: userSchema,
+	giftRecipient: giftRecipientSchema.nullable(),
+	product: productTypeSchema,
+	analyticsInfo: analyticsInfoSchema,
+	//TODO: This should probably be a date but the scala lambdas struggle to deserialise the default date representation
+	// so leave it as a string until all the lambdas are Typescript
+	firstDeliveryDate: z.string().nullable(),
```

Comment:
not sure what it is now or what the "default" date representation is, but for consistency it would be nice to be a zuora date format if that's what TS dates are compatible with e.g. "2025-02-03".
I remember for ease of deserialisation somewhere someone used (milli?)seconds since 1970 in a cookie we set, and it was a massive pain when you wanted to debug things.